Parse/src/main/java/com/parse/ParseACL.java

```diff
@@ -103,32 +103,41 @@ public static void setDefaultACL(ParseACL acl, boolean withAccessForCurrentUser)
     return getDefaultACLController().get();
   }
 
+  // State
+  private final Map<String, Permissions> permissionsById = new HashMap<>();
   private boolean shared;
+
   /**
    * A lazy user that hasn't been saved to Parse.
    */
   //TODO (grantland): This should be a list for multiple lazy users with read/write permissions.
   private ParseUser unresolvedUser;
 
-  private Map<String, Permissions> permissionsById;
-
   /**
    * Creates an ACL with no permissions granted.
    */
   public ParseACL() {
-    permissionsById = new HashMap<>();
+    // do nothing
   }
 
-  /* package */ ParseACL copy() {
-    ParseACL copy = new ParseACL();
-    for (String id : permissionsById.keySet()) {
-      copy.permissionsById.put(id, new Permissions(permissionsById.get(id)));
+  /**
+   * Creates a copy of {@code acl}.
+   *
+   * @param acl
+   *          The acl to copy.
+   */
+  public ParseACL(ParseACL acl) {
```

Comment:
It implicitly NPE on the next line, which I think is fine since Java APIs work similarly: https://android.googlesource.com/platform/libcore/+/refs/heads/master/luni/src/main/java/java/util/HashMap.java#194